Mathlib/Data/Bitvec/Lemmas.lean

```diff
@@ -124,4 +130,72 @@ theorem ofFin_toFin {n} (v : BitVec n) : ofFin (toFin v) = v := by
   rfl
 #align bitvec.of_fin_to_fin Std.BitVec.ofFin_toFin
 
+/-!
+### Distributivity of `Std.BitVec.ofFin`
+-/
+section
+variable (x y : Fin (2^w))
+
+@[simp] lemma ofFin_neg : ofFin (-x) = -(ofFin x) := by
+  rw [neg_eq_zero_sub]; rfl
+
+@[simp] lemma ofFin_and : ofFin (x &&& y) = ofFin x &&& ofFin y := rfl
+@[simp] lemma ofFin_or  : ofFin (x ||| y) = ofFin x ||| ofFin y := rfl
+@[simp] lemma ofFin_xor : ofFin (x ^^^ y) = ofFin x ^^^ ofFin y := rfl
+@[simp] lemma ofFin_add : ofFin (x + y)   = ofFin x + ofFin y   := rfl
+@[simp] lemma ofFin_sub : ofFin (x - y)   = ofFin x - ofFin y   := rfl
+@[simp] lemma ofFin_mul : ofFin (x * y)   = ofFin x * ofFin y   := rfl
+
+@[simp] lemma ofFin_zero    : @ofFin w 0 = 0 := rfl
+@[simp] lemma ofFin_one     : @ofFin w 1 = 1 := rfl
+
+end
+
+/-!
+### Distributivity of `Std.BitVec.toFin`
+-/
+section
+variable (x y : BitVec w)
+
+@[simp] lemma toFin_neg : toFin (-x) = -(toFin x) := by
+  rw [neg_eq_zero_sub]; rfl
+
+@[simp] lemma toFin_and : toFin (x &&& y) = toFin x &&& toFin y := rfl
+@[simp] lemma toFin_or  : toFin (x ||| y) = toFin x ||| toFin y := rfl
+@[simp] lemma toFin_xor : toFin (x ^^^ y) = toFin x ^^^ toFin y := rfl
+@[simp] lemma toFin_add : toFin (x + y)   = toFin x + toFin y   := rfl
+@[simp] lemma toFin_sub : toFin (x - y)   = toFin x - toFin y   := rfl
+@[simp] lemma toFin_mul : toFin (x * y)   = toFin x * toFin y   := rfl
+
+-- These should be simp, but Std's simp-lemmas do not allow this yet.
+lemma toFin_zero    : toFin (0 : BitVec w) = 0 := rfl
+lemma toFin_one     : toFin (1 : BitVec w) = 1 := rfl
+
+variable (n : Nat) (z : Int)
+
+lemma toFin_nsmul   : toFin (n • x) = n • x.toFin := rfl
+lemma toFin_zsmul   : toFin (z • x) = z • x.toFin := rfl
+@[simp] lemma toFin_pow     : toFin (x ^ n) = x.toFin ^ n := rfl
+@[simp] lemma toFin_natCast : @toFin w n = n              := rfl
```

Comment:
Can you add the `OfNat` lemma too? It will probably need to be after the ring instance

Reply:
I added these myself

Reply:
Thanks!